File: python/Programmers/예산.py
'''
정해진 예산으로 얼마나 많은 부서를 지원해줄 수 있는지 구하는 문제
부서가 원하는 예산을 정렬한 후 총 예산을 넘지 않는 구를 구하면 됨
'''

def solution(d, budget):
    answer = 0
    total = 0
    d.sort()
    
    for i in range(len(d)):
        total += d[i]
        
        if total > budget:
            break
            
        answer += 1
        
    return answer

File: Javascript/Programmers/예산.js
/*
가장 많은 부서에 예산을 나눠주려 할 때 최대 몇 부서한테 나눠줄 수 있는지 구하는 문제
*/

function solution(d, budget) {
    
  d.sort((a, b) => {
      return a - b
  })
  
  let maxDepartment = 0
  
  for (i = 0; i < d.length; i += 1) {
      if (budget - d[i] < 0) {
          break
      } else {
          maxDepartment += 1
          budget -= d[i]
      }
  }
  
  return maxDepartment;
}